### PR TITLE
Fix CLI entities command & add feature-views command

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: Existing CLI entities command is trying to use feast core. Instead, it should be using new FeastStore abstraction. Also, the CLI didn't have a way of accessing feature views, and I'm adding that as well. Here are the 4 new commands:
```
feast entities list
feast entities describe <entity-name>
feast feature-views list
feast feature-views describe <feature-view-name>
```

I also noticed that in bunch of places where we want to throw errors, we use `Exception` class, which has some disadvantages. For example, when you want to catch this error you have to use `except Exception as e`, which can catch all different kinds of exceptions. In this PR I started a new `errors.py` class where I introduces some new exception types, so it's easier to catch them.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Fix CLI entities command & add feature-views command
```

### CODE REVIEW

**Comment by woop** [comment]
> In this PR I started a new errors.py class where I introduces some new exception types, so it's easier to catch them.

Nice!

File: sdk/python/feast/cli.py [hunk 1]
```diff
@@ -118,57 +87,66 @@ def entity_describe(name: str, project: str):
...
+@feature_views_cmd.command("describe")
+@click.argument("name", type=click.STRING)
+def feature_view_describe(name: str):
+    """
+    Describe a feature view
+    """
+    cli_check_repo(Path.cwd())
+    store = FeatureStore(repo_path=str(Path.cwd()))
+
+    try:
```
Comment (woop) [inline sdk/python/feast/cli.py:123]: Is there a reason to have this try/except?

Comment (tsotnet) [reply sdk/python/feast/cli.py:123]: It's a difference between this
```
$ feast entities describe driver_id1
Traceback (most recent call last):
  File "/Users/tsotne/feast/venv/bin/feast", line 33, in <module>
    sys.exit(load_entry_point('feast', 'console_scripts', 'feast')())
  File "/Users/tsotne/feast/venv/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/tsotne/feast/venv/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/tsotne/feast/venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/tsotne/feast/venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/tsotne/feast/venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/tsotne/feast/venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/tsotne/feast/sdk/python/feast/cli.py", line 77, in entity_describe
    entity = store.get_entity(name)
  File "/Users/tsotne/feast/sdk/python/feast/feature_store.py", line 146, in get_entity
    return self._registry.get_entity(name, self.project)
  File "/Users/tsotne/feast/sdk/python/feast/registry.py", line 129, in get_entity
    raise EntityNotFoundException(project, name)
feast.errors.EntityNotFoundException: Entity driver_id1 does not exist in project my_repo
```
and this
```
$ feast entities describe driver_id1
Entity driver_id1 does not exist in project my_repo
```

I think having a stack trace is a worse user experience

Comment (woop) [reply sdk/python/feast/cli.py:123]: Yip, looks good to me.

**Review by woop — APPROVED** [review]

**Pull request merged**